jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/DefaultServlet.java

```diff
@@ -405,519 +67,47 @@ public String getInitParameter(String name)
-                    AsyncContext asyncContext = httpServletRequest.startAsync();
-                    Callback callback = new AsyncContextCallback(asyncContext, httpServletResponse);
-                    _resourceService.doGet(coreRequest, coreResponse, callback, content);
-                }
-                else
-                {
-                    // send the content blocking
-                    try (Blocker.Callback callback = Blocker.callback())
-                    {
-                        _resourceService.doGet(coreRequest, coreResponse, callback, content);
-                        callback.block();
-                    }
-                    catch (Exception e)
-                    {
-                        throw new ServletException(e);
-                    }
-                }
-            }
-        }
-        catch (InvalidPathException e)
-        {
-            if (LOG.isDebugEnabled())
-                LOG.debug("InvalidPathException for pathInContext: {}", encodedPathInContext, e);
-            if (included)
-                throw new FileNotFoundException(encodedPathInContext);
-            httpServletResponse.setStatus(404);
-        }
-    }
-
-    protected String getEncodedPathInContext(HttpServletRequest req, String includedServletPath)
-    {
-        if (includedServletPath != null)
-            return encodePath(getIncludedPathInContext(req, includedServletPath, !isDefaultMapping(req)));
-        else if (!isDefaultMapping(req))
-        {
-            //a match via an extension mapping will more than likely
-            //have no path info
-            String path = req.getPathInfo();
-            if (StringUtil.isEmpty(path) && 
-                MappingMatch.EXTENSION.equals(req.getHttpServletMapping().getMappingMatch()))
-                path = req.getServletPath();
-          
-            return encodePath(path);
-        }
-        else if (req instanceof ServletApiRequest apiRequest)
-            return Context.getPathInContext(req.getContextPath(), apiRequest.getRequest().getHttpURI().getCanonicalPath());
-        else
-            return Context.getPathInContext(req.getContextPath(), URIUtil.canonicalPath(req.getRequestURI()));
-    }
-
-    @Override
-    protected void doHead(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException
-    {
-        if (LOG.isDebugEnabled())
-            LOG.debug("doHead(req={}, resp={}) (calling doGet())", req, resp);
-        doGet(req, resp);
-    }
-
     @Override
-    protected void doTrace(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException
+    public void init() throws ServletException
     {
-        // Always return 405: Method Not Allowed for DefaultServlet
-        resp.sendError(HttpServletResponse.SC_METHOD_NOT_ALLOWED);
+        if ("true".equalsIgnoreCase(getInitParameter("pathInfoOnly")))
+            LOG.warn("DefaultServlet pathInfoOnly is set to true");
```

Comment:
```suggestion
            LOG.warn("DefaultServlet pathInfoOnly is set to true. Use ResourceServlet instead.");
```

Not 100% sure, but should we mention the ResourceServlet?